src/procstat.h

```diff
@@ -240,6 +240,25 @@ struct procstat_series_u64 {
 	uint64_t aggregated_variance;
 };
 
+
+/**
+ * @brief callback to calculate histogram values
+ */
+typedef void (*percentiles_calculator)(uint32_t *histogram,
+					uint64_t samples_count,
+					struct procstat_percentile_result *result,
+					unsigned result_len);
+
+#define MAX_SUPPORTED_PERCENTILE 20
+struct procstat_histogram_u32 {
+	uint64_t 			  sum;
+	uint64_t 			  count;
+	uint64_t 			  last;
+	int	 			  npercentile;
+	struct procstat_percentile_result percentile[MAX_SUPPORTED_PERCENTILE];
+	uint32_t 			  *histogram;
```

Comment:
@sashas-lb  nice work ! 

i guess the #1 usage for this is latency measurements 
for latency i would expect delta time in tsc as input 
(which B.T.W can be bigger than 32bits, while the code here expecting only 32bit values, anyhow the 32bit is probably fine for tsc delta...as we don;t really care for having accuracy in larger latency values)
 
but just to see if we're synced: 
do we convert time to other unit on data-path code...(i.e to  [us] or to other units)?...because conversion may be very costly,  IMHO if we think performance: we should avoid conversions on data-path, but only do it once wish to view the histogram (per user stats query...)

Make sense?